web/elm/src/Dashboard/Dashboard.elm

```diff
@@ -432,15 +447,24 @@ updatePipeline updater pipelineId model =
         | pipelines =
             model.pipelines
                 |> Maybe.map
-                    (List.Extra.updateIf
-                        (\p ->
-                            p.teamName == pipelineId.teamName && p.name == pipelineId.pipelineName
+                    (Dict.update pipelineId.teamName
+                        (Maybe.map
+                            (List.Extra.updateIf
+                                (\p -> p.name == pipelineId.pipelineName)
```

Comment:
ummm, I'm having trouble understand why we can drop the condition for `p.teamName == pipelineId.teamName` ...

Reply:
With this commit 9374d0cc8c6ebeb8c18381d963fb7d710f8f91b1, I changed how we stored pipelines internally. Previously, we stored them in a flat list like this:

```json
[
  {"teamName": "teamA", "pipelineName": "pipeline1"},
  {"teamName": "teamA", "pipelineName": "pipeline2"},
  {"teamName": "teamB", "pipelineName": "pipeline1"},
  {"teamName": "teamB", "pipelineName": "pipeline2"},
  {"teamName": "teamB", "pipelineName": "pipeline3"}
]
```

But now, we group them by team, like this:

```json
{
  "teamA": [
    {"teamName": "teamA", "pipelineName": "pipeline1"},
    {"teamName": "teamA", "pipelineName": "pipeline2"}
  ],
  "teamB": [
    {"teamName": "teamB", "pipelineName": "pipeline1"},
    {"teamName": "teamB", "pipelineName": "pipeline2"},
    {"teamName": "teamB", "pipelineName": "pipeline3"}
  ]
}
```

So, this function became:

```elm
(Dict.update pipelineId.teamName               -- Only update within the team's list of pipelines
    (Maybe.map
      (List.Extra.updateIf
        (\p -> p.name == pipelineId.pipelineName)
        updater
      )
    )
)
```

Since we're updating within the team's list of pipelines, each pipeline has the same team name, so we don't need to check it